### PR TITLE
add cloud run bug quirks-mode to getClientIP

### LINKED ISSUE #63: workaround broken cloud run X-Forwarded-For header
See details debugged in https://github.com/kubernetes-sigs/oci-proxy/pull/61#issuecomment-1105824535

/assign

been looking into the best forward plan for this, given it seems likely the bug will eventually be fixed.
will send a PR soonish hopefully.
/kind bug
/sig k8s-infra

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes #63

### CODE REVIEW

File: cmd/archeio/app/clientip_test.go [hunk 1]
```diff
@@ -63,6 +73,16 @@ func TestGetClientIP(t *testing.T) {
 			},
 			ExpectedIP: netip.MustParseAddr("8.8.8.8"),
 		},
+		{
+			Name: "X-Forwarded-For with clean client-supplied + cloud-run bug",
+			Request: http.Request{
+				Header: http.Header{
+					"X-Forwarded-For": []string{"127.0.0.1, 8.8.8.8, 8.8.8.9", "8.8.8.8"},
```
Comment (BenTheElder) [inline cmd/archeio/app/clientip_test.go:80]: /hold

**Comment by ameukam** [comment]
/lgtm
🤞🏾

File: cmd/archeio/app/clientip_test.go (hunk 1 above)
Comment (BenTheElder) [reply cmd/archeio/app/clientip_test.go]: comma should've been in the string, not in the slice, fixed

File: cmd/archeio/app/clientip.go [hunk 2]
```diff
@@ -47,7 +47,6 @@ func getClientIP(r *http.Request) (netip.Addr, error) {
 	// precede <client-ip>,<load-balancer-ip> in this header.
 	// The preceding IP addresses might contain other characters, including spaces.
 	rawXFwdFor := r.Header.Get("X-Forwarded-For")
```
Comment (BenTheElder) [inline cmd/archeio/app/clientip.go:49]: note: `Get` gets the first header value, I think cloud run flattens this header, but we will find out soon enough with some e2e tests.

Comment (BenTheElder) [reply cmd/archeio/app/clientip.go:49]: yes, `curl -v -H "Forwarded: aaa" -H "X-Forwarded-For: 8.8.8.8" -H "X-Forwarded-For: 127.0.0.1" https://registry-sandbox.k8s.io/v2/pause/blobs/sha256:da86e6ba6ca197bf6bc5e9d900febd906b133eaa4750e6bed647b0fbe50ed43aa` => logs show correct IP detected still.

**Pull request merged**